components/script/dom/webglrenderingcontext.rs

```diff
@@ -1518,29 +1530,25 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
 
     // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.2
     fn GetContextAttributes(&self) -> Option<WebGLContextAttributes> {
-        let (sender, receiver) = webgl_channel().unwrap();
-
-        // If the send does not succeed, assume context lost
-        if self
-            .webgl_sender
-            .send(WebGLCommand::GetContextAttributes(sender))
-            .is_err()
-        {
-            return None;
-        }
-
-        let attrs = receiver.recv().unwrap();
-
-        Some(WebGLContextAttributes {
-            alpha: attrs.alpha,
-            antialias: attrs.antialias,
-            depth: attrs.depth,
-            failIfMajorPerformanceCaveat: false,
-            preferLowPowerToHighPerformance: false,
-            premultipliedAlpha: attrs.premultiplied_alpha,
-            preserveDrawingBuffer: attrs.preserve_drawing_buffer,
-            stencil: attrs.stencil,
-        })
+    // Currently, user-specified attributes can only differ from the
+    // attributes of the generated context in depth, stencil and antialias.
+    // As of now though, antialias is simply ignored, and depth and
+    // stencil, when set to true, are assumed to always succeed, i.e.,
+    // there is no way the user-specified values will ever conflict
+    // with the ones we finally ended up with.
+    // Therefore, the values we had cached away when we instantiated
+    // the context stay the same, and can be safely returned back
+    // to the user.
+    return Some(WebGLContextAttributes {
```

Comment:
Maybe you won't need `return` here? Also, it would be great to indent one more tab inside the function!

```rust
fn GetContextAttributes(&self) -> Option<WebGLContextAttributes> {
    // Currently ...
    Some(WebGLContextAttributes {
        ...
    })
}
```

Reply:
Will do; just gonna wait for some other feedback and bundle up the changes if possible.